example/main.tf

```diff
@@ -0,0 +1,363 @@
+}
+
+resource "indykite_application" "development" {
+  # This is how it should be done properly, no count or other iteration
+  # See https://blog.gruntwork.io/terraform-tips-tricks-loops-if-statements-and-gotchas-f739bbae55f9
+  for_each     = {
+  for appSpace in data.indykite_application_spaces.more-of-them.app_spaces : appSpace.id => appSpace
+  }
+  app_space_id = each.key
+  name         = "app-for-${each.value.name}"
+  # deletion_protection=false
+}
+
+# ##########################
+# ### Application Agents ###
+# ##########################
+
+data "indykite_application_agent" "opa" {
+  app_space_id = data.indykite_application_space.wonka-factory.id
+  name         = "wonka-app-agent"
+}
+
+data "indykite_application_agents" "all-wonkas" {
+  # This is how it should be done properly, no count or other iteration
+  # See https://blog.gruntwork.io/terraform-tips-tricks-loops-if-statements-and-gotchas-f739bbae55f9
+  for_each     = {
+  for appSpace in data.indykite_application_spaces.more-of-them.app_spaces : appSpace.id => appSpace
+  }
+  app_space_id = each.key
+  filter       = ["wonka-app-agent", "non-existing-one", "loompaland-app-agent"]
+}
+
+output "app_agents" {
+  value       = flatten(values(data.indykite_application_agents.all-wonkas)[*].app_agents)
+  description = "Contains all application agents under multiple app spaces combined into single array"
+}
+
+resource "indykite_application_agent" "development" {
+  application_id = data.indykite_application.wonka-bars.id
+  name           = "app-agent-for-${data.indykite_application_space.wonka-factory.name}"
+  # deletion_protection=false
+}
+
+# ######################################
+# ### Application Agents Credentials ###
+# ######################################
+
+resource "indykite_application_agent_credential" "development" {
+  app_agent_id = data.indykite_application_agent.opa.id
+}
+
+resource "indykite_application_agent_credential" "with-public" {
+  app_agent_id      = data.indykite_application_agent.opa.id
+  display_name      = "Key with custom private-public key pair"
+  expire_time       = "2040-12-31T12:34:56-01:00"
+  default_tenant_id = data.indykite_tenant.wonka-1.id
+  public_key_jwk    = <<-EOT
+  {
+    "kty": "EC",
+    "use": "sig",
+    "crv": "P-256",
+    "kid": "sig-1631087814",
+    "x": "WZQ9LMC08l-L05oxRa-9ObmaPQlTuWHX2GaAmMgAuSE",
+    "y": "V5VuhYvyQ2ACiVznB_aqzfVWmwfhvPFD6Dc4X32WUE8",
+    "alg": "ES256"
+  }
+  EOT
+}
+
+//// Creates a new OAuth2 authenticator
+//resource "indykite_oauth2_client" "facebook" {
+//  tenant = indykite_tenant
+//
+//  name = "oauth2-facebook"
+//  description = "Facebook Authenticator"
+//
+//  client {
+//    provider = "facebook.com"
+//    client_id = "test"
+//    client_secret = "secret"
+//    redirect_uri = [
+//      "https://indykite-dev.indykite.id/handler/oauth2/callback"]
+//  }
+//
+//  depends_on = [
+//    indykite_service_authentication
+//  ]
+//}
+//
+//
```

Comment:
remove comments if not relevant

Reply:
Keeping for now, might use in future sprints when testing. If not, will remove then